diff -aurN SuperTuxKart-0.8.orig/lib/irrlicht/include/IrrCompileConfig.h SuperTuxKart-0.8/lib/irrlicht/include/IrrCompileConfig.h
--- SuperTuxKart-0.8.orig/lib/irrlicht/include/IrrCompileConfig.h	2013-04-16 11:26:27.374917675 +0200
+++ SuperTuxKart-0.8/lib/irrlicht/include/IrrCompileConfig.h	2013-04-16 11:31:36.175555468 +0200
@@ -245,7 +245,7 @@
 //! Define _IRR_USE_NON_SYSTEM_JPEG_LIB_ to let irrlicht use the jpeglib which comes with irrlicht.
 /** If this is commented out, Irrlicht will try to compile using the jpeg lib installed in the system.
 	This is only used when _IRR_COMPILE_WITH_LIBJPEG_ is defined. */
-#define _IRR_USE_NON_SYSTEM_JPEG_LIB_
+//#define _IRR_USE_NON_SYSTEM_JPEG_LIB_
 #ifdef NO_IRR_USE_NON_SYSTEM_JPEG_LIB_
 #undef _IRR_USE_NON_SYSTEM_JPEG_LIB_
 #endif
@@ -261,7 +261,7 @@
 //! Define _IRR_USE_NON_SYSTEM_LIBPNG_ to let irrlicht use the libpng which comes with irrlicht.
 /** If this is commented out, Irrlicht will try to compile using the libpng installed in the system.
 	This is only used when _IRR_COMPILE_WITH_LIBPNG_ is defined. */
-#define _IRR_USE_NON_SYSTEM_LIB_PNG_
+//#define _IRR_USE_NON_SYSTEM_LIB_PNG_
 #ifdef NO_IRR_USE_NON_SYSTEM_LIB_PNG_
 #undef _IRR_USE_NON_SYSTEM_LIB_PNG_
 #endif
@@ -598,7 +598,7 @@
 /** If this is commented out, Irrlicht will try to compile using the zlib
 installed on the system. This is only used when _IRR_COMPILE_WITH_ZLIB_ is
 defined. */
-#define _IRR_USE_NON_SYSTEM_ZLIB_
+//#define _IRR_USE_NON_SYSTEM_ZLIB_
 #ifdef NO_IRR_USE_NON_SYSTEM_ZLIB_
 #undef _IRR_USE_NON_SYSTEM_ZLIB_
 #endif
@@ -619,7 +619,7 @@
 /** If this is commented out, Irrlicht will try to compile using the bzlib
 installed on the system. This is only used when _IRR_COMPILE_WITH_BZLIB_ is
 defined. */
-#define _IRR_USE_NON_SYSTEM_BZLIB_
+//#define _IRR_USE_NON_SYSTEM_BZLIB_
 #ifdef NO_IRR_USE_NON_SYSTEM_BZLIB_
 #undef _IRR_USE_NON_SYSTEM_BZLIB_
 #endif
diff -aurN SuperTuxKart-0.8.orig/lib/irrlicht/source/Irrlicht/Makefile SuperTuxKart-0.8/lib/irrlicht/source/Irrlicht/Makefile
--- SuperTuxKart-0.8.orig/lib/irrlicht/source/Irrlicht/Makefile	2013-04-16 11:26:27.362917247 +0200
+++ SuperTuxKart-0.8/lib/irrlicht/source/Irrlicht/Makefile	2013-04-16 11:32:56.665648908 +0200
@@ -46,22 +46,17 @@
 IRRIOOBJ = CFileList.o CFileSystem.o CLimitReadFile.o CMemoryFile.o CReadFile.o CWriteFile.o CXMLReader.o CXMLWriter.o CWADReader.o CZipReader.o CPakReader.o CNPKReader.o CTarReader.o CMountPointReader.o irrXML.o CAttributes.o lzma/LzmaDec.o
 IRROTHEROBJ = CIrrDeviceSDL.o CIrrDeviceLinux.o CIrrDeviceConsole.o CIrrDeviceStub.o CIrrDeviceWin32.o CIrrDeviceFB.o CLogger.o COSOperator.o Irrlicht.o os.o
 IRRGUIOBJ = CGUIButton.o CGUICheckBox.o CGUIComboBox.o CGUIContextMenu.o CGUIEditBox.o CGUIEnvironment.o CGUIFileOpenDialog.o CGUIFont.o CGUIImage.o CGUIInOutFader.o CGUIListBox.o CGUIMenu.o CGUIMeshViewer.o CGUIMessageBox.o CGUIModalScreen.o CGUIScrollBar.o CGUISpinBox.o CGUISkin.o CGUIStaticText.o CGUITabControl.o CGUITable.o CGUIToolBar.o CGUIWindow.o CGUIColorSelectDialog.o CDefaultGUIElementFactory.o CGUISpriteBank.o CGUIImageList.o CGUITreeView.o
-ZLIBOBJ = zlib/adler32.o zlib/compress.o zlib/crc32.o zlib/deflate.o zlib/inffast.o zlib/inflate.o zlib/inftrees.o zlib/trees.o zlib/uncompr.o zlib/zutil.o
-JPEGLIBOBJ = jpeglib/jcapimin.o jpeglib/jcapistd.o jpeglib/jccoefct.o jpeglib/jccolor.o jpeglib/jcdctmgr.o jpeglib/jchuff.o jpeglib/jcinit.o jpeglib/jcmainct.o jpeglib/jcmarker.o jpeglib/jcmaster.o jpeglib/jcomapi.o jpeglib/jcparam.o jpeglib/jcprepct.o jpeglib/jcsample.o jpeglib/jctrans.o jpeglib/jdapimin.o jpeglib/jdapistd.o jpeglib/jdatadst.o jpeglib/jdatasrc.o jpeglib/jdcoefct.o jpeglib/jdcolor.o jpeglib/jddctmgr.o jpeglib/jdhuff.o jpeglib/jdinput.o jpeglib/jdmainct.o jpeglib/jdmarker.o jpeglib/jdmaster.o jpeglib/jdmerge.o jpeglib/jdpostct.o jpeglib/jdsample.o jpeglib/jdtrans.o jpeglib/jerror.o jpeglib/jfdctflt.o jpeglib/jfdctfst.o jpeglib/jfdctint.o jpeglib/jidctflt.o jpeglib/jidctfst.o jpeglib/jidctint.o jpeglib/jmemmgr.o jpeglib/jmemnobs.o jpeglib/jquant1.o jpeglib/jquant2.o jpeglib/jutils.o jpeglib/jcarith.o jpeglib/jdarith.o jpeglib/jaricom.o
-LIBPNGOBJ = libpng/png.o libpng/pngerror.o libpng/pngget.o libpng/pngmem.o libpng/pngpread.o libpng/pngread.o libpng/pngrio.o libpng/pngrtran.o libpng/pngrutil.o libpng/pngset.o libpng/pngtrans.o libpng/pngwio.o libpng/pngwrite.o libpng/pngwtran.o libpng/pngwutil.o
 LIBAESGM = aesGladman/aescrypt.o aesGladman/aeskey.o aesGladman/aestab.o aesGladman/fileenc.o aesGladman/hmac.o aesGladman/prng.o aesGladman/pwd2key.o aesGladman/sha1.o aesGladman/sha2.o
-BZIP2OBJ = bzip2/blocksort.o bzip2/huffman.o bzip2/crctable.o bzip2/randtable.o bzip2/bzcompress.o bzip2/decompress.o bzip2/bzlib.o
 
 # Next variable is for additional scene nodes etc. of customized Irrlicht versions
 EXTRAOBJ =
 LINKOBJ = $(IRRMESHOBJ) $(IRROBJ) $(IRRPARTICLEOBJ) $(IRRANIMOBJ) \
 	$(IRRVIDEOOBJ) $(IRRSWRENDEROBJ) $(IRRIOOBJ) $(IRROTHEROBJ) \
-	$(IRRGUIOBJ) $(ZLIBOBJ) $(JPEGLIBOBJ) $(LIBPNGOBJ) $(LIBAESGM) \
-	$(BZIP2OBJ) $(EXTRAOBJ)
+	$(IRRGUIOBJ) $(LIBAESGM) $(EXTRAOBJ)
 
 ###############
 #Compiler flags
-CXXINCS = -I../../include -Izlib -Ijpeglib -Ilibpng
+CXXINCS = -I../../include
 CPPFLAGS += $(CXXINCS) -DIRRLICHT_EXPORTS=1
 CXXFLAGS += -Wall -pipe -fno-exceptions -fno-rtti -fstrict-aliasing
 ifndef NDEBUG
diff -aurN SuperTuxKart-0.8.1.orig/CMakeLists.txt SuperTuxKart-0.8.1/CMakeLists.txt
--- SuperTuxKart-0.8.1.orig/CMakeLists.txt	2013-11-26 22:16:24.009803741 +0100
+++ SuperTuxKart-0.8.1/CMakeLists.txt	2013-11-26 22:19:05.701016004 +0100
@@ -222,14 +222,16 @@
 
 # Common library dependencies
 target_link_libraries(supertuxkart
+    ${IRRLICHT_LIBRARIES}
+    ${IRRLICHT_LIBRARY}
     bulletdynamics
     bulletcollision
     bulletmath
     enet
     stkirrlicht
-    ${CURL_LIBRARIES}
+    ${CURL_LIBRARIES} -lz
     ${OGGVORBIS_LIBRARIES}
-    ${IRRLICHT_XF86VM_LIBRARY}
+    ${IRRLICHT_XF86VM_LIBRARY} -ljpeg -lpng -lz -lbz2
     ${OPENAL_LIBRARY}
     ${OPENGL_LIBRARIES})
 
diff -aurN SuperTuxKart-0.8.1.orig/lib/irrlicht/CMakeLists.txt SuperTuxKart-0.8.1/lib/irrlicht/CMakeLists.txt
--- SuperTuxKart-0.8.1.orig/lib/irrlicht/CMakeLists.txt	2013-11-26 22:16:23.899799514 +0100
+++ SuperTuxKart-0.8.1/lib/irrlicht/CMakeLists.txt	2013-11-26 22:26:44.775653983 +0100
@@ -650,108 +650,6 @@
 include/IBoneSceneNode.h
 include/IShaderConstantSetCallBack.h
 include/ITexture.h
-
-
-source/Irrlicht/zlib/infback.c
-source/Irrlicht/zlib/deflate.c
-source/Irrlicht/zlib/zutil.c
-source/Irrlicht/zlib/crc32.c
-source/Irrlicht/zlib/gzread.c
-source/Irrlicht/zlib/inffast.c
-source/Irrlicht/zlib/uncompr.c
-source/Irrlicht/zlib/inftrees.c
-source/Irrlicht/zlib/inflate.c
-source/Irrlicht/zlib/compress.c
-source/Irrlicht/zlib/adler32.c
-source/Irrlicht/zlib/gzwrite.c
-source/Irrlicht/zlib/trees.c
-source/Irrlicht/zlib/gzlib.c
-source/Irrlicht/zlib/gzclose.c
-source/Irrlicht/zlib/inffixed.h
-source/Irrlicht/zlib/inffast.h
-source/Irrlicht/zlib/zconf.h
-source/Irrlicht/zlib/zutil.h
-source/Irrlicht/zlib/inftrees.h
-source/Irrlicht/zlib/deflate.h
-source/Irrlicht/zlib/zlib.h
-source/Irrlicht/zlib/inflate.h
-source/Irrlicht/zlib/trees.h
-source/Irrlicht/zlib/gzguts.h
-source/Irrlicht/zlib/crc32.h
-
-
-source/Irrlicht/jpeglib/jaricom.c
-source/Irrlicht/jpeglib/jcapimin.c
-source/Irrlicht/jpeglib/jcapistd.c
-source/Irrlicht/jpeglib/jcarith.c
-source/Irrlicht/jpeglib/jccoefct.c
-source/Irrlicht/jpeglib/jccolor.c
-source/Irrlicht/jpeglib/jcdctmgr.c
-source/Irrlicht/jpeglib/jchuff.c
-source/Irrlicht/jpeglib/jcinit.c
-source/Irrlicht/jpeglib/jcmainct.c
-source/Irrlicht/jpeglib/jcmarker.c
-source/Irrlicht/jpeglib/jcmaster.c
-source/Irrlicht/jpeglib/jcomapi.c
-source/Irrlicht/jpeglib/jcparam.c
-source/Irrlicht/jpeglib/jcprepct.c
-source/Irrlicht/jpeglib/jcsample.c
-source/Irrlicht/jpeglib/jctrans.c
-source/Irrlicht/jpeglib/jdapimin.c
-source/Irrlicht/jpeglib/jdapistd.c
-source/Irrlicht/jpeglib/jdarith.c
-source/Irrlicht/jpeglib/jdatadst.c
-source/Irrlicht/jpeglib/jdatasrc.c
-source/Irrlicht/jpeglib/jdcoefct.c
-source/Irrlicht/jpeglib/jdcolor.c
-source/Irrlicht/jpeglib/jddctmgr.c
-source/Irrlicht/jpeglib/jdhuff.c
-source/Irrlicht/jpeglib/jdinput.c
-source/Irrlicht/jpeglib/jdmainct.c
-source/Irrlicht/jpeglib/jdmarker.c
-source/Irrlicht/jpeglib/jdmaster.c
-source/Irrlicht/jpeglib/jdmerge.c
-source/Irrlicht/jpeglib/jdpostct.c
-source/Irrlicht/jpeglib/jdsample.c
-source/Irrlicht/jpeglib/jdtrans.c
-source/Irrlicht/jpeglib/jerror.c
-source/Irrlicht/jpeglib/jfdctflt.c
-source/Irrlicht/jpeglib/jfdctfst.c
-source/Irrlicht/jpeglib/jfdctint.c
-source/Irrlicht/jpeglib/jidctflt.c
-source/Irrlicht/jpeglib/jidctfst.c
-source/Irrlicht/jpeglib/jidctint.c
-source/Irrlicht/jpeglib/jmemmgr.c
-source/Irrlicht/jpeglib/jmemnobs.c
-source/Irrlicht/jpeglib/jquant1.c
-source/Irrlicht/jpeglib/jquant2.c
-source/Irrlicht/jpeglib/jutils.c
-
-
-source/Irrlicht/libpng/pngwutil.c
-source/Irrlicht/libpng/pngget.c
-source/Irrlicht/libpng/pngrtran.c
-source/Irrlicht/libpng/pngrutil.c
-source/Irrlicht/libpng/pngerror.c
-source/Irrlicht/libpng/pngset.c
-source/Irrlicht/libpng/pngtest.c
-source/Irrlicht/libpng/pngmem.c
-source/Irrlicht/libpng/pngpread.c
-source/Irrlicht/libpng/png.c
-source/Irrlicht/libpng/pngread.c
-source/Irrlicht/libpng/pngwio.c
-source/Irrlicht/libpng/pngtrans.c
-source/Irrlicht/libpng/pngwrite.c
-source/Irrlicht/libpng/pngwtran.c
-source/Irrlicht/libpng/example.c
-source/Irrlicht/libpng/pngrio.c
-source/Irrlicht/libpng/png.h
-source/Irrlicht/libpng/pngdebug.h
-source/Irrlicht/libpng/pngconf.h
-source/Irrlicht/libpng/pngpriv.h
-source/Irrlicht/libpng/pnginfo.h
-source/Irrlicht/libpng/pnglibconf.h
-source/Irrlicht/libpng/pngstruct.h
 )
 
 if(APPLE)
